features/dm.ts

```diff
@@ -367,3 +392,32 @@ function filterDMBetween(myPubKey: string, contactPubKey: string) {
             ((iAmAuthor && otherIsReceiver) || (otherIsAuthor && iAmReceiver));
     };
 }
+
+const cache = new Map<string, string | Error>();
```

Comment:
Technically this `cache` should live in `NostrAccountContext` but then I need to implement 2 both the in-memory ctx and the nip-7 ctx.

This is one of the rare cases where a global mutable object is more useful.